velox/type/Type.cpp

```diff
@@ -56,6 +56,7 @@ const std::unordered_map<std::string, TypeKind>& getTypeStringMap() {
       {"VARCHAR", TypeKind::VARCHAR},
       {"VARBINARY", TypeKind::VARBINARY},
       {"TIMESTAMP", TypeKind::TIMESTAMP},
+      {"DATE", TypeKind::INTEGER},
```

Comment:
Is this change correct? Should this map include logical types? It feels like probably no.

Reply:
There were a few places using (try)mapNameToTypeKind functions where isDate calls needed to be added. It felt cleaner to add this case here then.

e.g. https://github.com/facebookincubator/velox/blob/main/velox/expression/FunctionSignature.cpp#L169

https://github.com/facebookincubator/velox/blob/main/velox/substrait/TypeUtils.cpp#L75

https://github.com/facebookincubator/velox/blob/main/velox/expression/tests/ArgumentTypeFuzzer.cpp#L57

Reply:
The problem with adding logical types here is that we cannot do that for logical types that are not built-in, e.g. added by an application.

Reply:
We may need to look into these places to see how they handle custom types.

Reply:
Ok. I'll remove this for now.

Reply:
> We may need to look into these places to see how they handle custom types.

Its possible they don't handle custom types in general.

Reply:
Have removed that cleanup so that non-built-in logical types can be handled outside that code.